features/cellular/framework/AT/AT_CellularDevice.cpp

```diff
@@ -163,6 +163,24 @@ nsapi_error_t AT_CellularDevice::get_sim_state(SimState &state)
     return error;
 }
 
+nsapi_error_t AT_CellularDevice::set_puk(const char *puk, const char *sim_pin)
+{
+    if (sim_pin == NULL || puk == NULL) {
+        return NSAPI_ERROR_PARAMETER;
+    }
+
+    _at->lock();
+    _at->set_at_timeout(5000);
```

Comment:
if timeout is changed, it must be set back with _at->restore_at_timeout();

Reply:
@jarvte I don't think so. The prototype of the function is  `set_at_timeout(uint32_t timeout, bool default_timeout = false) `. So by default it wouldn't store the timeout as default.

Reply:
Ah, it doesn't restore after use.  The doc needs to be added perhaps.

Reply:
cmd_stop_read_resp() method should restore the timeout logically if default_timeout was false in set_at_timeout(...)